### ISSUE TITLE
how to pass additional args to kubelet when running  'k3s agent'

### ISSUE THREAD

**Issue by huaixichen** [k3s-io/k3s#190]
any help would be appreciated!

**Comment by seiyaa** [comment]
k3s agent --help?not work?

**Comment by huaixichen** [comment]
> o

i am looking for a way to specify --pod-infra-container-image of kubelet. it seems no such option in 'k3s agent'

**Comment by huaixichen** [comment]
well, digging into the codes, it seems no initialization of 'ExtraKubeletArgs' and 'ExtraKubeProxyArgs' in 'Agent' structure. 

```
type Agent struct {
	NodeName           string
	ClusterCIDR        net.IPNet
	ClusterDNS         net.IP
	RootDir            string
	KubeConfig         string
	NodeIP             string
	RuntimeSocket      string
	ListenAddress      string
	CACertPath         string
	CNIBinDir          string
	CNIConfDir         string
	ExtraKubeletArgs   []string
	ExtraKubeProxyArgs []string
}
```

**Comment by brandond** [comment]
Fixed by #309
